### DISCUSSION TITLE
Mouse action "keybinding" - assign functionality per Godot editor view

### DISCUSSION THREAD

**Discussion by clockworkprince** [godotengine/godot-proposals#4750] [Editor]
I've been looking for a way to swap the functionality of the middle and right mouse buttons when viewing 3D scenes, but I could not find any way to do so. If this is truly something that is not currently possible, I would like to request that this gets implemented. It would also be nice if this was per view, so 2D scenes can have their mouse functionality altered as well. 

Right now, when pressing the right mouse button in 3D view, the editor's view of the 3D scene gets panned. The middle mouse button rotates the view around a specific central point. I would like to swap these two functionalities. 

My main arguments for this being added as a feature is that it would improve accessibility for the Godot editor and it is probably not very complicated to implement.

**Comment by Calinou** [comment]
See also https://github.com/godotengine/godot-proposals/issues/3881.

Rebinding individual buttons used in 3D navigation isn't possible yet, but there are a few schemes you can select. That said, none of them match the settings you're requesting, so it may be better to scrap those schemes entirely and allow for using arbitrary actions instead.

In the 2D editor, you can hold <kbd>Space</kbd> and use the left mouse button to pan the view (similar to most image editors). Using the right mouse button to pan will be possible again if https://github.com/godotengine/godot/pull/58371 is merged.

In the 3D editor, when you need to use freelook for extended periods of time, *toggle* it with <kbd>Shift + F</kbd> instead of holding down the right mouse button. Press <kbd>Shift + F</kbd> again or <kbd>Escape</kbd> to exit freelook mode after toggling it.

**Reply by clockworkprince** [reply]
`Shift+F` does not toggle anything for me when I press it in the 3D editor. I don't know what it's supposed to do, but it changes nothing about the way the mouse behaves for me. It doesn't swap the functionality of right click and middle click, nor does it change what left click and moving the mouse does. (I use v3.4.4 on Windows.)

The linked thread also doesn't seem to be asking for quite the same thing, because I also want middle mouse butting rebinding, not just right mouse button.

**Reply by Calinou** [reply]
> Shift+F does not toggle anything for me when I press it in the 3D editor.

You need to be currently focusing on the 3D viewport for it to work. When the 3D viewport is focused, you'll see a faint blue line bordering the 3D viewport (or whatever accent color you've chosen in the editor settings).

**Reply by clockworkprince** [reply]
> > Shift+F does not toggle anything for me when I press it in the 3D editor.
> 
> You need to be currently focusing on the 3D viewport for it to work. When the 3D viewport is focused, you'll see a faint blue line bordering the 3D viewport (or whatever accent color you've chosen in the editor settings).

I managed to make it work, but all it does is look around, but not pivot around the centred point in the scene. It's not really a replacement for what I'm looking for, unfortunately.
